internal/runners/export/log.go

```diff
@@ -0,0 +1,84 @@
+package export
+
+import (
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+)
+
+type Log struct {
+	output.Outputer
+}
+
+func NewLog(prime primeable) *Log {
+	return &Log{prime.Output()}
+}
+
+type LogParams struct {
+	Prefix string
+	Index  int
+}
+
+type logFile struct {
+	Name      string
+	Timestamp int
+}
+
+var ErrInvalidLogIndex = errs.New("invalid index")
+var ErrInvalidLogPrefix = errs.New("invalid log prefix")
+var ErrLogNotFound = errs.New("log not found")
+
+func (l *Log) Run(params *LogParams) (rerr error) {
+	defer rationalizeError(&rerr)
+
+	if params.Index < 0 {
+		return ErrInvalidLogIndex
+	}
+	if params.Prefix == "" {
+		params.Prefix = "state"
+	}
+
+	// Fetch list of log files.
+	logDir := filepath.Dir(logging.FilePath())
+	logFiles := fileutils.ListDirSimple(logDir, false)
+
+	// Filter down the list based on the given prefix.
+	filteredLogFiles := []*logFile{}
+	regex, err := regexp.Compile(params.Prefix + `-\d+-(\d+)\.log`)
+	if err != nil {
+		return ErrInvalidLogPrefix
+	}
+	for _, file := range logFiles {
+		if regex.MatchString(file) {
+			timestamp, err := strconv.Atoi(regex.FindStringSubmatch(file)[1])
```

Comment:
Does the `regex.MatchString` guarantee that we will find a second group? If not the indexing of the slice here is a potential panic.

Reply:
Yes, if there is a match, the capture from the pattern guarantees a valid index.